src/test/ui/discrim/discrim-ill-typed.stderr

```diff
@@ -3,88 +3,48 @@ error[E0308]: mismatched types
    |
 LL |         OhNo = 0_u8,
    |                ^^^^ expected `i8`, found `u8`
-   |
-help: change the type of the numeric literal from `u8` to `i8`
-   |
-LL |         OhNo = 0_i8,
-   |                ^^^^
 
 error[E0308]: mismatched types
   --> $DIR/discrim-ill-typed.rs:30:16
    |
 LL |         OhNo = 0_i8,
    |                ^^^^ expected `u8`, found `i8`
-   |
-help: change the type of the numeric literal from `i8` to `u8`
```

Comment:
This seems like an unfortunate regression... should we exclude enum discriminants from the early return?

Reply:
The reason is because we have an early return to avoid incorrectly suggest `.into()` on `const`s, and because of it we actually have some fixes in some cases. I can tackle cleaning up `check_for_cast` in a separate subsequent PR.

Reply:
Fixed in this PR.